File: c/26.c
#include "stdio.h"

int removeDuplicates(int* nums, int numsSize) {
    if (0 == numsSize)
        return 0;
    int pos = 1;
    int now = nums[0];
    int i = 0;
    int m;
    for (i = 1; i < numsSize; i++){
        m = nums[i];
        if (m > now){
            now = m;
            nums[pos++] = now;
        }
    }
    return pos;
}

int main(){
    int nums[3] = {1,1,2};
    printf("%d\n",removeDuplicates(nums,3));
}

File: python/26.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.append('a')
        i = 0
        while True:
            if nums[i] == 'a':
                break
            if nums[i] == nums[i+1]:
                del nums[i+1]
            else:
                i += 1
        return i


s = Solution()
n1 = [1,1,2]
print s.removeDuplicates(n1)